controllers/cluster_controller.go

```diff
@@ -241,3 +233,20 @@ func (r *ClusterReconciler) listChildren(ctx context.Context, cluster *clusterv1
 
 	return children, nil
 }
+
+func (r *ClusterReconciler) patchCluster(ctx context.Context, cluster *clusterv1.Cluster, patch client.Patch) error {
+	log := r.Log.WithValues("cluster-namespace", cluster.Namespace, "cluster-name", cluster.Name)
+	// TODO(chuckha): .Patch will use the typed client and in the case of a noop on the Spec will return an empty status which is automatically assigned to cluster.
+	// This means that Status.Patch will not update anything since all new Status data was eliminated.
+	statusCopy := cluster.Status.DeepCopy()
+	if err := r.Client.Patch(ctx, cluster, patch); err != nil {
+		log.Error(err, "Error patching cluster")
+		return err
+	}
+	statusCopy.DeepCopyInto(&cluster.Status)
+	if err := r.Client.Status().Patch(ctx, cluster, patch); err != nil {
```

Comment:
It's actually a good thing that cluster is updated. If you use a deep copy here then the object going into Status().Patch will end up being out-of-date since it was updated in the Patch and the update will fail.

Reply:
It shouldn't affect patch the same way as update did, since it will not hit the resourceVersion issue.